microservice/subscription/src/test/java/com/cumulocity/microservice/subscription/repository/impl/LegacyMicroserviceRepositoryTest.java

```diff
@@ -320,6 +322,102 @@ public void shouldGetSubscriptions(){
                 .isSameAs(user);
     }
 
+    @Test
+    public void shouldCreateAppWithExtensionsIfMetadataContains() {
+        repository.register(
+                getMetadataWithExtensions());
+
+        assertThat(platform.take(byMethod(POST)))
+                .hasSize(1)
+                .extracting("body")
+                .have(new Condition<Object>() {
```

Comment:
condition could be extracted to self describing method in order to improve readibility